### PR TITLE
Awake the blocking callers when Tune(size int) is invoked to expand the pool

### LINKED ISSUE #205: 当使用Tune(size int)放大容量时，之前blocking的任务并不会立即执行
当使用Tune(size int)放大容量时，之前blocking的任务并不会立即执行
经过测试，正在运行的一个任务结束后才会按照当前被Tune后的真实cap去执行任务

这里想问一下除了我在外坚挺cap变化后再提交新任务这种方法外，还有没有别的方法可以让我在放大容量后立即执行blocking的任务

### PR DESCRIPTION
**Pull request by codingfanlt**
…capacity

---
name: Pull request
about: Propose changes to the code
title: 'Instantly scale up capacity when using Tune(size int) to enlarge the capacity'
labels: 'enhancement'
assignees: ''
---

## 1. Are you opening this pull request for bug-fixs, optimizations or new feature?

When using Tune(size int) to enlarge the capacity, create a worker immediately to perform blocking tasks.

## 3. Please link to the relevant issues (if any).

close #205 

## 4. Checklist

- [x] I have squashed all insignificant commits.
- [x] I have commented my code for explaining package types, values, functions, and non-obvious lines.
- [x] I have written unit tests and verified that all tests passes (if needed).
- [ ] I have documented feature info on the README (only when this PR is adding a new feature).
- [x] (optional) I am willing to help maintain this change if there are issues with it later.

### CODE REVIEW

File: pool.go [hunk 1]
```diff
@@ -180,10 +180,18 @@ func (p *Pool) Cap() int {
...
+	if capacity == -1 || size <= 0 || size == capacity || p.options.PreAlloc {
 		return
 	}
 	atomic.StoreInt32(&p.capacity, int32(size))
+	if size > capacity {
+		if size-capacity == 1 {
+			p.cond.Signal()
+			return
+		}
+		p.cond.Broadcast()
```
Comment (panjf2000) [inline pool.go:193]: 麻烦补充一点测试。

**Review by panjf2000 — CHANGES_REQUESTED** [review]

File: pool.go (hunk 1 above)
Comment (codingfanlt) [reply pool.go:193]: > 麻烦补充一点测试。

已添加相关单元测试

Comment (panjf2000) [reply pool.go:193]: 这部分修改也需要改 pool_func.go

Comment (codingfanlt) [reply pool.go:193]: 已修改

File: ants_test.go [hunk 2]
```diff
@@ -670,3 +670,71 @@ func TestRestCodeCoverage(t *testing.T) {
...
+				<-c
+			})
+		}()
+	}
+	p.Tune(8)
+	wg.Wait()
+	if n := p.Running(); n != 8 {
+		t.Errorf("expect 8 workers running, but got %d", n)
+	}
+
```
Comment (panjf2000) [inline ants_test.go:709]: 这里加一下 `close(c)`，然后 `p.Release()`，下面同理。

**Review by panjf2000 — CHANGES_REQUESTED** [review]

File: ants_test.go (hunk 2 above)
Comment (codingfanlt) [reply ants_test.go:713]: 已修改，但其实 `p.Release()` 之后仍然有 task 在跑。如果想要优雅关闭需要添加 `pool.Wait()` 方法在程序退出前等待 task 全部运行完毕。

**Review by panjf2000 — APPROVED** [review]
LGTM

**Comment by panjf2000** [comment]
Thanks~

**Pull request merged**